com.spotify.Client.json

```diff
@@ -150,6 +150,88 @@
                 }
             ]
         },
+        /* THESE CANNOT BE UPDATED UNTIL SPOTIFY SUPPORTS ffmpeg 5 ------------------------------------------------------- */
+        {
+            "name": "sdl2",
+            "buildsystem": "simple",
+            "build-commands": [
+                "cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX:PATH=/app -B Build",
+                "cd Build && make -j${FLATPAK_BUILDER_N_JOBS:-1} && make install"
+            ],
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://github.com/libsdl-org/SDL/releases/download/release-2.0.22/SDL2-2.0.22.tar.gz",
+                    "sha256": "fe7cbf3127882e3fc7259a75a0cb585620272c51745d3852ab9dd87960697f2e"
+                }
+            ]
+        },
+        {
+            "name": "ffmpeg-4.4",
+            "build-options": {
+                "cflags": "-O2 -g -fstack-protector-strong -D_FORTIFY_SOURCE=2",
+                "cxxflags": "-O2 -g -fstack-protector-strong -D_FORTIFY_SOURCE=2",
+                "ldflags": "-fstack-protector-strong -Wl,-z,relro,-z,now",
+                "prefix": "/app/",
+                "env": {
+                    "V": "1"
+                }
+            },
+            "config-opts": [
+                /* Copied from runtine 21.08 */
```

Comment:
I think this overcomplicates things as runtime build is very specific. It may be better just copy this from another app that uses ffmpeg or from spotify git history

Reply:
That was just a note i made to note where i got the build flags are borrowed from. They have been tweaked a little but i believe they should be fine for this.

Reply:
My point is those buildflags are copied without reason behind them. Runtime passes `--disable-everything` then lists every --enable=xxx, codec,muxer etc. explicitly  - you don't pass `--disable-everything` but still list them all. This doesn't make sense and won't be understandable by future contributors.

Reply:
ok so should i just remove `--disable-everything` and leave then rest in then?

Reply:
> My point is those buildflags are copied without reason behind them. Runtime passes `--disable-everything` then lists every --enable=xxx, codec,muxer etc. explicitly - you don't pass `--disable-everything` but still list them all. This doesn't make sense and won't be understandable by future contributors.

This is a mistake. i will add `--disable-everything` back. it was removed because i was having issues compiling (see sdl2) and forgot to add it back

Reply:
But why do you think you need to copy those options? 90% of them are unnecessary.

Reply:
> > My point is those buildflags are copied without reason behind them. Runtime passes `--disable-everything` then lists every --enable=xxx, codec,muxer etc. explicitly - you don't pass `--disable-everything` but still list them all. This doesn't make sense and won't be understandable by future contributors.
> 
> This is a mistake. i will add `--disable-everything` back. it was removed because i was having issues compiling (see sdl2) and forgot to add it back

I might be wrong about my reason. give me a secons

Reply:
Yes i was wrong. It was removed because the flag disables something that prevents spotify local files from working. using default flags and then making sure the rest is manually enabled is the best solution in my opinion.
Feel free to correct me.

Reply:
Some non-default things may need to be enabled and same default things may need to be disabled (like encoders since they aren't useful). I don't want to ask you to audit every ffmpeg option but I think going back to what was [known to work](https://github.com/flathub/com.spotify.Client/commit/554b766ee55434d2eb35612e674d0b1dfce3ac9f) is easiest solution. Just copy the old config and retain the version you already have here.

Reply:
ok i will do that and test.

Reply:
> Some non-default things may need to be enabled and same default things may need to be disabled (like encoders since they aren't useful). I don't want to ask you to audit every ffmpeg option but I think going back to what was [known to work](https://github.com/flathub/com.spotify.Client/commit/554b766ee55434d2eb35612e674d0b1dfce3ac9f) is easiest solution. Just copy the old config and retain the version you already have here.

The reason i didnt do that in the first place was because i didnt realize spotify used to have ffmpeg in the first place. My mistake. ill sort that now just give me a min

Reply:
Done